src/dnaio/multipleend.py

```diff
@@ -0,0 +1,221 @@
+import contextlib
+from os import PathLike
+from typing import BinaryIO, IO, Iterable, Iterator, List, Optional, Tuple, Union
+
+from xopen import xopen
+
+from ._core import SequenceRecord, records_are_mates
+from .exceptions import FileFormatError
+from .interfaces import MultipleFileWriter
+from .readers import FastaReader, FastqReader
+from .singleend import _open_single, _detect_format_from_name
+from .writers import FastaWriter, FastqWriter
+
+
+def _open_multiple(
+    *files: Union[str, PathLike, BinaryIO],
+    fileformat: Optional[str] = None,
+    mode: str = "r",
+    qualities: Optional[bool] = None,
+    opener=xopen,
+):
+    if not files:
+        raise ValueError("at least one file is required")
+    if mode not in ("r", "w", "a"):
+        raise ValueError("mode must be one of 'r', 'w', 'a'")
+    elif mode == "r":
+        return MultipleFileReader(*files, fileformat=fileformat, opener=opener)
+    elif mode == "w" and fileformat is None:
+        # Assume mixed files will not be offered.
+        for file in files:
+            if isinstance(file, str):
```

Comment:
Should this also be made to work if `file` is `PathLike`?

Reply:
That is quite an easy thing. Done.